apps/docs/components/ProjectConfigVariables.tsx

```diff
@@ -0,0 +1,455 @@
+}
+
+const instanceStore = proxy({
+  selectedInstanceKey: null as InstanceKey | null,
+  setSelectedInstanceKey: (instanceKey: InstanceKey | null) => {
+    instanceStore.selectedInstanceKey = instanceKey
+    if (instanceKey !== null) {
+      store('local', LOCAL_STORAGE_KEYS.SAVED_ORG_PROJECT_BRANCH, toStoredString(instanceKey))
+    }
+  },
+  clear: () => {
+    instanceStore.setSelectedInstanceKey(null)
+  },
+})
+
+type Variable = 'url' | 'anonKey'
+
+const prettyFormatVariable: Record<Variable, string> = {
+  url: 'Project URL',
+  anonKey: 'Anon key',
+}
+
+type ProjectConfigVariablesState =
+  | 'userLoading'
+  | 'loggedOut'
+  | 'loggedIn.dataPending'
+  | 'loggedIn.hasData'
+  | 'loggedIn.hasNoData'
+  | 'loggedIn.dataError'
+
+type ProjectConfigVariablesPlusApiState =
+  | Omit<ProjectConfigVariablesState, 'loggedIn.hasData'>
+  | 'loggedIn.hasData.apiDataPending'
+  | 'loggedIn.hasData.apiDataSuccess'
+  | 'loggedIn.hasData.apiDataError'
+
+function ComboBox({
+  parentStateSummary,
+  instances,
+  className,
+}: {
+  parentStateSummary: ProjectConfigVariablesState
+  instances: InstanceData[]
+  className?: string
+}) {
+  const { selectedInstanceKey, setSelectedInstanceKey } = useSnapshot(instanceStore)
+  const selectedInstance = getInstanceDataFromKey(instances, selectedInstanceKey)
+
+  const [open, setOpen] = useState(false)
+
+  return (
+    <Popover open={open} onOpenChange={setOpen}>
+      <div className={cn('flex items-center', className)}>
+        <PopoverTrigger asChild aria-label={`Select project`}>
+          <Button
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            className="overflow-hidden h-auto min-h-10 flex justify-between border-none px-1 text-left"
+          >
+            {parentStateSummary === 'userLoading' || parentStateSummary === 'loggedIn.dataPending'
+              ? 'Loading...'
+              : toPrettyString(selectedInstance) || 'Select a project...'}
+            <ChevronsUpDown className="ml-2 h-4 w-4 shrink-0 opacity-50" />
+          </Button>
+        </PopoverTrigger>
+      </div>
+      <PopoverContent className="z-[999999] p-0" side="bottom">
+        <Command>
+          <CommandInput placeholder="Search project..." className="border-none ring-0" />
+          <CommandEmpty>No project found.</CommandEmpty>
+          <CommandGroup>
+            {instances.map((instance) => (
+              <CommandItem
+                key={toStoredString(toInstanceKey(instance))}
+                onSelect={(selectedValue: string) => {
+                  const newSelectedInstance = fromPrettyString(instances, selectedValue)
+                  setSelectedInstanceKey(toInstanceKey(newSelectedInstance))
+                  setOpen(false)
+                }}
+              >
+                <Check
+                  className={cn(
+                    'mr-2 h-4 w-4',
+                    instance === selectedInstance ? 'opacity-100' : 'opacity-0'
```

Comment:
selected state probably should be based on project ref to account for identical project names?

![image](https://github.com/supabase/supabase/assets/19742402/acad0560-4696-4852-a3c6-b9506783e873)